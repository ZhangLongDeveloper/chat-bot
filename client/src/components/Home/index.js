import React from 'react';

import EmptyView from '../Common/EmptyView';

const DESCRIPTION = `
  This is home
`;

const Home = () => (
  <div>
    <EmptyView description={DESCRIPTION} />
  </div>
);

export default Home;
