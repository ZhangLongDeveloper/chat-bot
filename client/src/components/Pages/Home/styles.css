.home-page-container .basic-info-section {
  display: flex;
  flex-direction: row;
  padding: 50px 25px 25px 25px;
}

.home-page-container .basic-info-section .left-column {
  text-align: center;
  margin-right: 25px;
}

.home-page-container .my-profile-photo {
  width: 250px;
  border-radius: 5px;
  box-shadow: 1px 1px 10px grey;
}

@media (max-width: 750px) {
  .home-page-container .my-profile-photo {
    width: 150px;
  }
}

@media (max-width: 500px) {
  .home-page-container .my-profile-photo {
    width: 200px;
  }

  .home-page-container .basic-info-section {
    flex-direction: column;
  }
}