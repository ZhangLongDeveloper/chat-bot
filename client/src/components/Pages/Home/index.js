import React from 'react';

import EmptyView from '../../Common/EmptyView';
import { myProfilePicture } from '../../../images';

import './styles.css';

const Home = () => {
  const renderDescription = () => (
    <React.Fragment>
      <p>
        Welcome to my new personal project!!!
      </p>

      <p>
        Ask questions as you are speaking to me and try to populate
        the website with the info he will provide.
      </p>

      <p>
        Every page will have a progress bar which indicates
        how much info you have already recovered - so you will know
        where to stop asking about some particular topic.
      </p>

      <p>
        One more time, this project was created to learn about Chatbots
        and to practice my full stack skills. I am happy at my current job and
        I am not looking for a change.
      </p>
    </React.Fragment>
  );

  const renderInfo = () => (
    <React.Fragment>
      <div className="basic-info-section">
        <div className="left-column">
          <img src={myProfilePicture} alt="Me" className="my-profile-photo" />
        </div>
        <div className="right-column">
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            Nunc turpis metus, lacinia dapibus ex non, ultricies ornare quam.
            Aliquam erat volutpat. Morbi convallis, purus ut vehicula rhoncus,
            mauris magna condimentum mauris, eget facilisis eros nulla sed justo.
            Aliquam fermentum eros ut tortor gravida finibus.
          </p>
        </div>
      </div>

      <div className="summary-section">
        <h3>About me:</h3>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          Nunc turpis metus, lacinia dapibus ex non, ultricies ornare quam.
          Aliquam erat volutpat. Morbi convallis, purus ut vehicula rhoncus,
          mauris magna condimentum mauris, eget facilisis eros nulla sed justo.
          Aliquam fermentum eros ut tortor gravida finibus.
        </p>
      </div>

      <div className="languages-section">
        <h3>Languages I speak:</h3>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          Nunc turpis metus, lacinia dapibus ex non, ultricies ornare quam.
          Aliquam erat volutpat. Morbi convallis, purus ut vehicula rhoncus,
          mauris magna condimentum mauris, eget facilisis eros nulla sed justo.
          Aliquam fermentum eros ut tortor gravida finibus.
        </p>
      </div>

      <div className="hobbies-section">
        <h3>My hobbies:</h3>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          Nunc turpis metus, lacinia dapibus ex non, ultricies ornare quam.
          Aliquam erat volutpat. Morbi convallis, purus ut vehicula rhoncus,
          mauris magna condimentum mauris, eget facilisis eros nulla sed justo.
          Aliquam fermentum eros ut tortor gravida finibus.
        </p>
      </div>
      
    </React.Fragment>
  );

  return (
    <div className="home-page-container">
      {false && <EmptyView section="home">{renderDescription()}</EmptyView>}
      {true && renderInfo()}
    </div>
  );
};

export default Home;
