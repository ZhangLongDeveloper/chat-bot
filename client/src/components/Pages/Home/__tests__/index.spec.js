import React from 'react';
import { create } from 'react-test-renderer';

import Home from '../../Home';

jest.mock('../../../Common/EmptyView', () => 'EmptyView');

const mockComponent = props => (
  <Home {...props} />
);

describe('components/pages/<Home />', () => {
  it('should render component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
