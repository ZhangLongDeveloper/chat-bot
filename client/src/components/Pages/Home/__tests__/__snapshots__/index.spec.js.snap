// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Pages/<Home /> should render component 1`] = `
<div
  className="home-page-container"
>
  <div
    className="basic-info-section"
  >
    <div
      className="left-column"
    >
      <img
        alt="Me"
        className="my-profile-photo"
        src="test-file-stub"
      />
    </div>
    <div
      className="right-column"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc turpis metus, lacinia dapibus ex non, ultricies ornare quam. Aliquam erat volutpat. Morbi convallis, purus ut vehicula rhoncus, mauris magna condimentum mauris, eget facilisis eros nulla sed justo. Aliquam fermentum eros ut tortor gravida finibus.
      </p>
    </div>
  </div>
  <div
    className="summary-section"
  >
    <h3>
      About me:
    </h3>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc turpis metus, lacinia dapibus ex non, ultricies ornare quam. Aliquam erat volutpat. Morbi convallis, purus ut vehicula rhoncus, mauris magna condimentum mauris, eget facilisis eros nulla sed justo. Aliquam fermentum eros ut tortor gravida finibus.
    </p>
  </div>
  <div
    className="languages-section"
  >
    <h3>
      Languages I speak:
    </h3>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc turpis metus, lacinia dapibus ex non, ultricies ornare quam. Aliquam erat volutpat. Morbi convallis, purus ut vehicula rhoncus, mauris magna condimentum mauris, eget facilisis eros nulla sed justo. Aliquam fermentum eros ut tortor gravida finibus.
    </p>
  </div>
  <div
    className="hobbies-section"
  >
    <h3>
      My hobbies:
    </h3>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc turpis metus, lacinia dapibus ex non, ultricies ornare quam. Aliquam erat volutpat. Morbi convallis, purus ut vehicula rhoncus, mauris magna condimentum mauris, eget facilisis eros nulla sed justo. Aliquam fermentum eros ut tortor gravida finibus.
    </p>
  </div>
</div>
`;
